### PR TITLE
Make sixpack.Session take an object of options instead of args.

### PR DESCRIPTION
**Pull request by HeyHugo**
Some motivation:
Now the order of arguments matters which makes the API a bitt stiff.

Example now you write undefined if you only want to set base_url:
```
sixpack.Session(undefined, 'http://mysixpackserver.com');
```

New API:
```
sixpack.Session({base_url: 'http://mysixpackserver.com'});
```

### CODE REVIEW

**Comment by HeyHugo** [comment]
Hmm this is a breaking API change...
If you want I can change so that you can use either an options object argument or the old ones, would that be better?

**Comment by zackkitzmiller** [comment]
@HeyHugo I'm happy with a major version bump and shipping this.

**Comment by HeyHugo** [comment]
Ok cool!
(Sorry I forgot to branch off from master before making this pull request.)

I've made another feature that I'd like to put in another PR and see if you like it.

**Comment by zackkitzmiller** [comment]
Sounds good to me. However, JS isn't my strong suit, so I'd like to have a colleague review if you don't mind. @zsherman would you mind?

File: sixpack.js [hunk 1]
```diff
@@ -18,15 +18,16 @@
...
+        options = options || {};
+        this.client_id = options.client_id || sixpack.generate_client_id();
+        this.base_url = options.base_url || sixpack.base_url;
+        this.ip_address = options.ip_addess || sixpack.ip_address;
+        this.user_agent = options.user_agent || sixpack.user_agent;
         if (!on_node) {
             this.user_agent = this.user_agent || (window && window.navigator && window.navigator.userAgent);
         }
-        this.timeout = timeout || sixpack.timeout;
+        this.timeout = options.timeout || sixpack.timeout;
```
Comment (zsherman) [inline sixpack.js]: What happens if the timeout is 0? This pattern for setting defaults isn't great, we should refactor it to use some version of ```extends```

**Comment by HeyHugo** [comment]
@zsherman I agree, extends/assign would be better.

What about using Object.assign with this polyfill? https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Polyfill

**Comment by zsherman** [comment]
:+1:

**Comment by HeyHugo** [comment]
@zsherman Any feedback on my latest commits?

**Comment by zsherman** [comment]
@HeyHugo looks like you dropped the default options altogether, unless I'm missing something?

**Comment by HeyHugo** [comment]
Nope it's still there https://github.com/HeyHugo/sixpack-js/blob/master/sixpack.js#L5
or d you mean something else?

**Comment by zsherman** [comment]
Ah sorry totally missed that, looks good.

**Comment by HeyHugo** [comment]
:+1:

**Comment by mdressman** [comment]
doesn't look like the version bump ever happened? @HeyHugo @zackkitzmiller @zsherman

**Pull request merged**